src/site/antora/modules/ROOT/examples/manual/lookups/MainArgsExample.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public final class MainArgsExample implements Runnable {
+
+    // tag::usage[]
+    private final Logger logger = LogManager.getLogger(); // <1>
+
+    public static void main(final String[] args) {
+        try { // <2>
+            Class.forName("org.apache.logging.log4j.core.lookup.MainMapLookup")
+                    .getDeclaredMethod("setMainArguments", String[].class)
+                    .invoke(null, (Object) args);
+        } catch (final ReflectiveOperationException e) {
+            // Log4j Core is not used.
+        }
```

Comment:
Why don't we directly call `MainMapLookup.setMainArguments()`? I mean, if you use `setMainArguments()`, you do this to access it in the configuration, which implies you are already using Log4j Core. I am not able to see the point of indirection here. What am I missing? What is the use case addressed by this indirection?

Reply:
I am not a big fan of applications with a hardcoded logging backend.

If a user wishes to switch the logging implementation, he should be able to do it by simply replacing the runtime dependencies.

**Disclaimer**: Personally I don't see a lot of applications for the `main` lookup. Most applications will actually parse command line options. If the application has some logging-related options, it will export them to another lookup.

For example Spring Boot export `LOG_FILE` as a system property.

Reply:
Okay, makes sense, I guess. :+1: